### PR TITLE
Introduce a new tool for Catching unhandled Python exception in Docker

### PR DESCRIPTION
**Pull request by mhabrnal**
More info in blog post: https://abrt.github.io/openshift/abrt/python/docker/2018/02/08/abrt-for-docker/

### CODE REVIEW

**Comment by jfilak** [comment]
Why do we need a completely the new file abrt_exception_handler3_container.py without any use of the exiting infrastructure?

I would teach the tool abrt-container-logger to pass its entire input (STDIN) to the fd. Passing error messages on command line parameters is error prone as you can pass only 65535 bytes :)

**Comment by xsuchy** [comment]
> Why do we need a completely the new file abrt_exception_handler3_container.py without any use of the exiting infrastructure?

Because we assume there is nothing in the container. And we want to modify it as less as possible. Handler and setuid wrapper is this minimal set. We are able to retrieve this information on host and show it OS dashboard.

> I would teach the tool abrt-container-logger to pass its entire input (STDIN) to the fd. Passing error messages on command line parameters is error prone as you can pass only 65535 bytes :)

Good point.

File: abrt.spec.in [hunk 1]
```diff
@@ -309,6 +309,16 @@ Obsoletes: abrt-addon-python3 < 2.10.4
 This package contains python 3 hook and python analyzer plugin for handling
 uncaught exception in python 3 programs.
 
+%package -n python3-abrt-container-addon
+Summary: %{name}'s container addon for catching Python 3 exceptions
+Provides: abrt-container-addon-python3 = %{version}-%{release}
+Provides: abrt-container-addon-python3%{?_isa} = %{version}-%{release}
```
Comment (xsuchy) [inline abrt.spec.in]: Why these provides?

File: abrt.spec.in [hunk 2]
```diff
@@ -309,6 +309,16 @@ Obsoletes: abrt-addon-python3 < 2.10.4
...
+Conflicts: python3-abrt-addon
+
+%description -n python3-abrt-container-addon
+This package contains python 3 hook and for handling
```
Comment (xsuchy) [inline abrt.spec.in]: s/and //

File: src/hooks/abrt_exception_handler3_container.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    else:
+        # We don't know the path.
+        # (BTW, we *can't* assume the script is in current directory.)
+        executable = sys.argv[0]
+
+    data = dict()
+    data["pid"] = os.getpid()
+    data["executable"] = executable
+    data["reason"] = tb_text.splitlines()[0]
+    data["backtrace"] = tb_text
```
Comment (xsuchy) [inline src/hooks/abrt_exception_handler3_container.py]: Why not:
data = {
  "pid": os.getpid(),
   "executable": executable,
...
}
```

**Comment by mhabrnal** [comment]
Addressed all your comments

**Comment by xsuchy** [comment]
I would suggest to put that binary to /usr/libexec and execute it with full path.

**Pull request merged**